server/events/vcs/github_client.go

```diff
@@ -430,16 +481,85 @@ func (g *GithubClient) UpdateStatus(ctx context.Context, request types.UpdateSta
 }
 
 // [WENGINES-4643] TODO: Move the checks implementation to UpdateStatus once github checks is stable
-// UpdateChecksStatus updates the status check
 func (g *GithubClient) UpdateChecksStatus(ctx context.Context, request types.UpdateStatusRequest) error {
-	// TODO: Implement updating github checks
-	// - Get all checkruns for this SHA
-	// - Match the UpdateReqIdentifier with the check run. If it exists, update the checkrun. If it does not, create a new check run.
+	checkRuns, err := g.GetRepoChecks(request.Repo, request.Ref)
+	if err != nil {
+		return err
+	}
 
-	// Checks uses Status and Conlusion. Need to map models.CommitStatus to Status and Conclusion
-	// Status -> queued, in_progress, completed
-	// Conclusion -> failure, neutral, cancelled, timed_out, or action_required. (Optional. Required if you provide a status of "completed".)
-	return nil
+	status, conclusion := g.resolveChecksStatus(request.State)
+	checkRunOutput := github.CheckRunOutput{
+		Title:   &request.StatusName,
+		Summary: &request.Description,
+	}
+	if request.Output != "" {
+		checkRunOutput.Text = &request.Output
+	}
+
+	for _, checkRun := range checkRuns {
+		if *checkRun.Name == request.StatusName {
```

Comment:
If we always update one check run we can flatten this loop. Into something like this 
```
if checkRun := findCheckRun(request.StatusName, checkRuns); checkRun != nil {
// do checkrun update
} else {
// do checkrun create
}
```